app/models/UserAnswers.scala

```diff
@@ -59,6 +59,17 @@ final case class UserAnswers(
       case Left(errors) => Left(errors)
     }
 
+  def getOptionalPageValueForOptionalBooleanPage[A](
+    answers: UserAnswers,
+    questionPage: QuestionPage[Boolean],
+    optionalPage: QuestionPage[A]
+  )(implicit rds: Reads[A]): EitherNec[ValidationError, Option[A]] =
+    getPageValue(questionPage) match {
+      case Right(true)  => getPageValue(optionalPage).map(Some(_))
```

Comment:
Could this be written like this:

`case Right(true)  => getPageValue(optionalPage).map(Some(_))
case _      => unexpectedValueDefined(answers, optionalPage)`

Reply:
Done